### ISSUE TITLE
webpack error due to "jQuery" not being all lowercase

### ISSUE THREAD

**Issue by mikembm** [farbelous/bootstrap-colorpicker#272]
require("jQuery") should be require("jquery")

When using "jQuery" webpack throws the following error:
"There are multiple modules with names that only differ in casing"

This error occurs when you require other modules which also utilize jquery (i.e. jquery-ui)

**Comment by itsjavi** [comment]
thanks for reporting @mikembm 
can you tell me in which file and line is this problem coming from? is it coming from bootstrap-colorpicker.js dist file?

I don't know if the problem may come from the webpack config

```js
new webpack.ProvidePlugin({
      $: 'jquery',
      jQuery: 'jquery', // change this to jquery: 'jquery', ?
      'window.$': 'jquery',
      'window.jQuery': 'jquery'
    }),
```

Can you help me reproducing the issue with an example code? thanks

**Comment by th3conc3pt3ur** [comment]
Hi here, i've got the same problem. I install this with npm.
For resolution i change in /node_modules/bootstrap-colorpicker/dist/js/bootstrap-colorpicker.js all occurence of "JQuery" to "jquery".
Hope it's help :/

Here the console error message :

```
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* ....\node_modules\jQuery\dist\jquery.js
    Used by 1 module(s), i. e.
    .....\node_modules\bootstrap-colorpicker\dist\js\bootstrap-colorpicker.js
* ....\node_modules\jquery\dist\jquery.js
    Used by 8 module(s), i. e.
```

**Comment by nafplann** [comment]
same errors here.. is there any plan to merge the fixes by @th3conc3pt3ur ?

**Comment by ada3m0n666** [comment]
same here and same fix.

**Comment by mathieupotier** [comment]
The issue occurs on system that are case sensitive ... (mainly UNIX systems)
when loaders are trying to resolve jQuery in the `node_modules` dir, it doesn't exists ...

**Comment by archonic** [comment]
The suggestion for providing a lowercase jquery with `webpack.ProvidePlugin` doesn't work. It looks like the review has gone well. Are there any other barriers to merging? I'm happy to help if I can.

**Comment by mathieupotier** [comment]
just approving the PR ... ^_^

**Comment by oliverhughes** [comment]
+1 Same issue here, no fix other than renaming 'jQuery' to 'jquery' in `bootstrap-colorpicker.js`, which is really inconvenient when doing scheduled server deploys...!

**Comment by 614mediagroup** [comment]
Ran into this issue as well. Thanks for the temporary fix tell the real fix gets pulled in.

**Comment by fefoweb** [comment]
Yes, also I have the same issue and i change /jQuery/jquery/g on the dist file... issues gone :)
Thanks for the fix!

**Comment by jaapz** [comment]
Would be great to have this fixed! It's currently blocking me from upgrading to the v3+ version

**Comment by piperone** [comment]
I'm also stuck on the 2.x branch until this is fixed.

If the fix is a breaking change, consider updating the major version number (i.e. following semantic versioning).

**Comment by alexandre-pereira** [comment]
We need this to be fixed please :)

**Comment by cosmini** [comment]
Same problem here  😢

**Comment by TD-Media** [comment]
Same problem here, solved it for now by renaming 'jQuery' to 'jquery' in bootstrap-colorpicker.js and moving the file to my own assets folder. I hope the real fix will get pulled in soon.

**Comment by luckylu** [comment]
Same here on macOS. Please fix it as soon as possible.Thanks

**Comment by icex** [comment]
+1 same issue

**Comment by mathieupotier** [comment]
Requested changes have been produced in this PR https://github.com/farbelous/bootstrap-colorpicker/pull/285

**Comment by Guymestef** [comment]
Can we have a release of this fix pushed on npm soon? 🙏

**Comment by itsjavi** [comment]
it's broken since @mathieupotier changes -> https://github.com/farbelous/bootstrap-colorpicker/pull/285#issuecomment-438204884

I am waiting for him or someone else to fix it, because I don't have time sorry.

**Comment by Star1122** [comment]
I am suffering from same issue on mac.
When will this issue be fixed?

**Comment by thomasdarde** [comment]
@mathieupotier Hi ! 
I'm a bit lost with the different PR, could you tell me the most recent one to work on ?

**Comment by koenhendriks** [comment]
Same issue here on mac with Laravel. Don't think npm repo is updated?

**Comment by itsjavi** [comment]
@thomasdarde @koenhendriks there is no PR anymore, the one that we merged was breaking the plugin on browsers.
This bug needs to be re-addressed and tested properly in both node and browsers.

**Comment by Mbilallayers** [comment]
Hi, How to fix webpack error due to "jQuery" not being all lowercase?

**Comment by dlxko** [comment]
NPM repo need 2be updated ? Still getting problems :(

**Comment by alfredopenaalonso** [comment]
@itsjavi, I've created a PR with the change which seems to solve the problem.

I've tested it, and after the build, only the first lines change `require("jQuery")` with `require("jquery")`.

**Comment by itsjavi** [comment]
@alfredopenaalonso I think it will break the plugin in the browser. We need a solution that works for both node environment and browser. But I am out of ideas :/

**Comment by jaapz** [comment]
Couldn't you set up two separate builds, one for browser, one for the version published to npm?

**Comment by itsjavi** [comment]
@ here 

This should be fixed in v3.1.1
